### PR TITLE
Jsonnet/Helm: make memcached connection limit configurable

### PR DESCRIPTION
**Pull request by lukas-unity**
#### What this PR does

It makes possible to set memcached connection limit, instead of using the hard coded 16384 limit

#### Checklist

- [ ] Tests updated.
- [ ] Documentation added.
- [X] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [ ] [`about-versioning.md`](/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

**Review by dimitarvdimitrov — APPROVED** [review]
LGTM, thank you!

**Comment by 56quarters** [comment]
> I see that jsonnet tests are failing, I assume that I should make the the variables unique, instead of using `memcached_cache_connection_limit` ? (never used jsonnet)

Yep. If you could match the existing variable naming too, that'd be great. Thanks!

**Comment by lukas-unity** [comment]
@dimitarvdimitrov sorry, I've made a typo with formatting the value, now it should be good and tests should pass

File: operations/mimir/memcached.libsonnet [hunk 1]
```diff
@@ -52,7 +52,7 @@ memcached {
         name: 'memcached-frontend',
         max_item_size: '%dm' % [$._config.cache_frontend_max_item_size_mb],
         overprovision_factor: 1.05,
-        connection_limit: 16384,
+        connection_limit: '%d' % [$._config.cache_frontend_connection_limit],
```
Comment (56quarters) [inline operations/mimir/memcached.libsonnet:55]:
```suggestion
        connection_limit: std.toString($._config.cache_frontend_connection_limit),
```

File: operations/mimir/memcached.libsonnet [hunk 2]
```diff
@@ -64,7 +64,7 @@ memcached {
         name: 'memcached-index-queries',
         max_item_size: '%dm' % [$._config.cache_index_queries_max_item_size_mb],
         overprovision_factor: 1.05,
-        connection_limit: 16384,
+        connection_limit: '%d' % [$._config.cache_index_queries_connection_limit],
```
Comment (56quarters) [inline operations/mimir/memcached.libsonnet:67]:
```suggestion
        connection_limit: std.toString($._config.cache_frontend_connection_limit),
```

File: operations/mimir/memcached.libsonnet [hunk 3]
```diff
@@ -79,7 +79,7 @@ memcached {
         // Save memory by more tightly provisioning memcached chunks.
         memory_limit_mb: 6 * 1024,
         overprovision_factor: 1.05,
-        connection_limit: 16384,
+        connection_limit: '%d' % [$._config.cache_chunks_connection_limit],
```
Comment (56quarters) [inline operations/mimir/memcached.libsonnet:82]:
```suggestion
        connection_limit: std.toString($._config.cache_frontend_connection_limit),
```

File: operations/mimir/memcached.libsonnet [hunk 4]
```diff
@@ -90,7 +90,7 @@ memcached {
       $.memcached {
         name: 'memcached-metadata',
         max_item_size: '%dm' % [$._config.cache_metadata_max_item_size_mb],
-        connection_limit: 16384,
+        connection_limit: '%d' % [$._config.cache_metadata_connection_limit],
```
Comment (56quarters) [inline operations/mimir/memcached.libsonnet:93]:
```suggestion
        connection_limit: std.toString($._config.cache_frontend_connection_limit),
```

**Review by 56quarters — APPROVED** [review]
LGTM, thanks!

**Pull request merged**